src/dune_rules/install_rules.ml

```diff
@@ -1001,12 +984,45 @@ let scheme_per_ctx_memo =
       let* schemes = Memo.Build.sequential_map packages ~f:(scheme sctx) in
       Scheme.evaluate ~union:Rules.Dir_rules.union (Scheme.all schemes))
 
-let gen_rules sctx ~dir =
-  let* rules, subdirs =
+let symlink_rules sctx ~dir =
+  let+ rules, subdirs =
     let* scheme = Memo.exec scheme_per_ctx_memo sctx in
     Scheme.Evaluated.get_rules scheme ~dir
   in
-  let+ () =
-    Rules.produce_dir ~dir (Option.value ~default:Rules.Dir_rules.empty rules)
+  ( Subdir_set.These subdirs
+  , match rules with
+    | None -> Rules.empty
+    | Some rules -> Rules.of_dir_rules ~dir rules )
+
+let gen_install_alias sctx (package : Package.t) =
+  let ctx = Super_context.context sctx in
+  let name = Package.name package in
+  if not ctx.implicit then
```

Comment:
if this `if` is inverted, it ill become easier to read.

Reply:
Did that